evm_loader/program/src/entrypoint.rs

```diff
@@ -185,15 +239,26 @@ fn process_instruction<'a>(
             if program_info.owner != program_id {
                 return Err(ProgramError::InvalidArgument);
             }
+
             do_write(program_info, offset, &bytes)
         },
+        EvmInstruction::WriteContractCode {offset, bytes} => {  
```

Comment:
Merge with `Write`. Add checks that user can change contract data.
1. empty account
2. contract without code (code_size = 0)